crates/ruff_linter/src/rules/pylint/rules/single_dispatch_method.rs

```diff
@@ -0,0 +1,71 @@
+use ruff_diagnostics::{Diagnostic, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast};
+use ruff_python_semantic::{analyze::function_type, Scope};
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for singledispatch decorators on class methods.
+///
+/// ## Why is this bad?
+/// Single dispatch must happen on the type of first non self argument
```

Comment:
Example code is helpful here. Take a look at other rules to see the format. The pylint docs contains examples to use. 

You can also link the Python documentation of singledispatchmethod